### ISSUE TITLE
Some search plugins show incorrect size

### ISSUE THREAD

**Issue by key-sir2329** [BurningMop/qBittorrent-Search-Plugins#16]
Hello, recently I've noticed that some search plugins no longer show the correct size of the torrent, it seems to truncate the size, for example something that is 27GiB will show as 27B. The following plugins are affected:

therarbg
torrentdownloadspro

Other search plugins like bitsearch and 1337x still function properly.

**Comment by key-sir2329** [comment]
here is a patch,  I don't have write access to the repo so I could not open a PR:
```
diff --git a/therarbg.py b/therarbg.py
index 4b31381..cd8f171 100644
--- a/therarbg.py
+++ b/therarbg.py
@@ -128,7 +128,7 @@ class therarbg(object):
                 self.shouldGetCategory = False

             if self.shouldGetSize:
-                self.row['size'] = data.replace(',', '.')
+                self.row['size'] = data.replace(',', '.').replace('\xa0', ' ')
                 self.shouldGetSize = False

             if self.shouldGetSeeds:
diff --git a/torrentdownloads.py b/torrentdownloads.py
index 39d7501..517c469 100644
--- a/torrentdownloads.py
+++ b/torrentdownloads.py
@@ -125,7 +125,7 @@ class torrentdownloads(object):
                 self.shouldParseName = False

             if self.shouldGetSize:
-                size = data.replace('&nbsp;', '')
+                size = data.replace('&nbsp;', '').replace('\xa0', ' ')
                 self.row['size'] = size
                 self.shouldGetSize = False
```

**Comment by BurningMop** [comment]
Hi, thanks for your contribution, I'm pretty sure you can create a PR. Bear in mind this is a non official repo, and while I have contributed to the qbittorrent codebase, these are not part of the official repo

**Comment by BurningMop** [comment]
Fixed with PR #17

**Comment by sisi399** [comment]
@BurningMop 

I’m not sure when this issue resurfaced, but I’ve been experiencing it with the **torrentdownloadspro** plugin for a while now. I’ve included a fix below (AI-generated - please double-check).

```
diff --git a/torrentdownloads.py.bak b/torrentdownloads.py
index ef57aa1..68d329a 100644
--- a/torrentdownloads.py.bak
+++ b/torrentdownloads.py
@@ -105,7 +105,7 @@ class torrentdownloads(object):
                 self.shouldParseName = True
                 href = params.get('href')
                 if href.startswith("/torrent/"):
-                    link = f'{self.url}/{href}'
+                    link = f'{self.url}{href}'
                     self.row['desc_link'] = link
 
                     torrent_page = retrieve_url(link)
@@ -124,9 +124,13 @@ class torrentdownloads(object):
                 self.shouldParseName = False
 
             if self.shouldGetSize:
-                size = data.replace('&nbsp;', '').replace('\xa0', ' ')
-                self.row['size'] = size
-                self.shouldGetSize = False
+                # PATCH: Remove non-breaking spaces and labels, then convert to Bytes
+                if "Total Size:" in data:
+                    return
+                size = data.replace('\xa0', ' ').replace('&nbsp;', ' ').strip()
+                if size:
+                    self.row['size'] = anySizeToBytes(size)
+                    self.shouldGetSize = False
 
             if self.shouldGetSeeds:    
                 self.row['seeds']  = data
@@ -145,7 +149,7 @@ class torrentdownloads(object):
 
             if tag == self.DIV and self.insideRow:
                 self.row['engine_url'] = self.url
-                if not self.shouldSkipResult:
+                if not self.shouldSkipResult and 'size' in self.row:
                     prettyPrinter(self.row)
                 self.column = 0
                 self.row = {}
@@ -186,4 +190,4 @@ class torrentdownloads(object):
             page += 1
 
         for t in threads:
-            t.join()
+            t.join()
\ No newline at end of file

```
